rust/lance-core/src/io/writer/statistics.rs

```diff
@@ -287,34 +877,29 @@ impl StatisticsBuilder {
     }
 
     fn boolean_appender(&mut self, row: StatisticsRow) {
-        self.null_count.append_value(row.null_count);
+        let ScalarValue::Int64(Some(null_count)) = row.null_count else {
+            todo!()
+        };
+        let ScalarValue::Boolean(Some(max_value)) = row.max_value else {
+            todo!()
+        };
+        let ScalarValue::Boolean(Some(min_value)) = row.min_value else {
+            todo!()
+        };
```

Comment:
Also needs fixing. That fact that this doesn't cause the test to fail indicates you need to improve test coverage here.

Reply:
Changed.